### DISCUSSION TITLE
Allow to customize `/admin` path

### DISCUSSION THREAD

**Discussion by viters** [directus/directus#17667] [Feature Requests]
### Summary

https://github.com/directus/directus/blob/main/api/src/app.ts#L184
https://github.com/directus/directus/blob/main/api/src/app.ts#L206

`admin` should be configurable with ENV.

### Motivation

It is often a good idea to run admin panel on path different than `/admin`.

### Must Have:
- `ADMIN_PATH` environment variable with default `admin`.

### Alternatives

None.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by vogdb** [reply]
@rijkvanzanten Hi! Will be this functionality considered if I make a proper PR for that?

**Reply by vogdb** [reply]
Ok, now I see where the problem is. Strictly saying there is no easy way to achieve this currently as this new env variable that holds the custom admin URI must be shared among the back (`api` folder) and the front part (`app` folder) of Directus. That is why `admin` path is hardcoded. It can be easily extracted in `api` folder but the same can't be applied to `app` folder. That is sad :(.

**Reply by vogdb** [reply]
To elaborate more here. Here is the commit that updates `api` folder for this new env https://github.com/friendlee-pro/directus/commit/11fc2feb7ab3c912f88f001e727367ea42401ab2. And here the commit for `app` https://github.com/friendlee-pro/directus/commit/6f7f5bd4999167f485ce504ad0dfd17c088a4738. @rijkvanzanten I'm still open to implement the PR if you are OK with it. I only need your suggestion on the architecture of such solution. How to make `app` to use env variable from `api`.

In general. Sorry for such fuss about this topic.

**Comment by vogdb** [comment]
Here is the nginx solution for anyone in trouble as me for Directus 10.7.2. It does not change `/admin` to `/custom`. It prohibits accessing `/admin` directly. Instead you have to go to `/custom` which in turn will redirect you to `/admin` internally. Also I suggest to use `/content` instead of `/custom`. This way you will be redirected correctly after the login. @rijkvanzanten please review if possible.

```nginx
server {
    listen 80;
    listen [::]:80;
    server_name example.com;

    # this is our magic door. Only via it you can access Admin panel.
    location /custom {
        proxy_pass http://localhost:8055/admin;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header REMOTE-HOST $remote_addr;
        add_header X-Cache $upstream_cache_status;

        proxy_connect_timeout 30s;
        proxy_read_timeout 86400s;
        proxy_send_timeout 30s;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }

    # this one handles all Directus routes except `/admin`
    location @redirect {
       proxy_pass http://localhost:8055;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header REMOTE-HOST $remote_addr;
        add_header X-Cache $upstream_cache_status;

        proxy_connect_timeout 30s;
        proxy_read_timeout 86400s;
        proxy_send_timeout 30s;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }

    location / {
        try_files /dev/null @redirect;
    }

    # this one blocks all direct requests to Admin panel so it is kind of hidden.
    location /admin {
        location ~* ^.+\.(?:css(\.map)?|js(\.map)?|jpe?g|png|gif|ico|cur|heic|webp|tiff?|mp3|m4a|aac|ogg|midi?|wav|mp4|mov|webm|mpe?g|avi|ogv|flv|wmv|svgz?|ttf|ttc|otf|eot|woff2?)$ {
            try_files /dev/null @redirect;
        }
        return 404;
    }
```

**Reply by rijkvanzanten** [reply]
> please review if possible.

I don't use Nginx myself so can't help you there
